tests/FormTest.php

```diff
@@ -11,6 +39,403 @@
  */
 class FormTest extends TestCase
 {
+    /**
+     * @expectedException        \RuntimeException
+     * @expectedExceptionMessage Duplicate component name 
+     */
+    public function testAddComponentShouldThrowRuntimeException()
+    {
+        $form = new Form();
+        $form->addButtonButton('submit');
+        $form->addButtonButton('submit');
+    }
+
+    public function testAddButtonButton()
+    {
+        $form = new Form();
+        $button = $form->addButtonButton('submit_button');
+
+        $this->assertInstanceOf(Button::class, $button);
+        $this->assertSame(['submit_button' => ''], $form->getValues());
+        $this->assertStringMatchesFormat('<form><button type="button" name="submit_button" id="%s"></button></form>', (string) $form);
+    }
+
+    public function testGetComponent()
+    {
+        $form = new Form();
+        $form->addButtonButton('submit');
+
+        $this->assertFalse($form->getComponent('submit')->isRequired());
```

Comment:
It would be more logical to test that $form->getComponent() returns an instanceof Button\Button